modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -805,8 +816,15 @@ public String getSchemaType(Schema p) {
             return openAPIType;
         } else if (typeMapping.containsKey(openAPIType)) {
             type = typeMapping.get(openAPIType);
+            String typeWithoutGeneric = null;
```

Comment:
please extract the `typeWithoutGeneric` calculation into a separate method to deduplicate the code